### ISSUE TITLE
associatedBondingCurve

### ISSUE THREAD

**Issue by aszx29** [chainstacklabs/pump-fun-bot#26]
Hello, I have a question I would like to consult with you about. Your program and nodes are great, and now what I need to do is improve my purchase speed. Although using logsSubscribe generates a high number of requests, its speed is much faster than blockSubscribe. However, using log subscription leads to the lack of associatedBondingCurve, but it can be generated by utilizing the captured information since it's a PDA. Does your team know which seeds and the PUMP_PROGRAM to use to generate it?

**Comment by akegaviar** [comment]
hey @aszx29 your question actually got me thinking so I spent my Sunday morning figuring it out 🙂 . Check the readme file in the root directory for the description and what's used for seeds.

TLDR I have implemented this in two `/learning-examples` scripts: 

* compute_associated_bonding_curve.py — computes the associated bonding curve for a given token address
* `listen_new_direct_full_details.py` — the same script as `listen_new_direct.py` that uses just `logsSubscribe` but now also prints the computed associatedBondngCurve address

Some time later I'll probably implement this into the main trading script, but as of now you should be fine anyway with these examples.

**Comment by aszx29** [comment]
Oh, I’m really sorry to disturb your rest time. Thank you so much for your explanation. The program is already great, and the current examples are more than enough. I also hope your company will continue to grow and reach even greater heights.

**Comment by akegaviar** [comment]
Hey there, starting from the next week (**week of March 10**) we'll be rolling out updates to the bot v2 based on the feedback and the reported issues, including updating to the latest libs, better error handling etc. We are already actively working on it.

That'll be in a separate branch.

Check out the readme for the roadmap that I just added in the README file for the full overview.

**Comment by smypmsa** [comment]
Hi @aszx29! Please check the refactored branch. This feature has been implemented there 😃 Feel free to share your feedback.

https://github.com/chainstacklabs/pump-fun-bot/tree/refactored/main-v2
